src/views/network/view_session.cpp

```diff
@@ -15,11 +15,18 @@ namespace big
 	{
 		static uint64_t rid = 0;
 		ImGui::InputScalar("Input RID", ImGuiDataType_U64, &rid);
-		components::button("Join RID", []
+		components::button("Join by RID", []
 		{
 			session::join_by_rockstar_id(rid);
 		});
 
+		static char username[20];
+		ImGui::InputText("Input Username", username, sizeof(username));
+		components::button("Join by Username", []
+		{
+			session::join_by_username(username);
+		});
```

Comment:
Probably best to use `g_thread_pool::push()` as the API lookup will definitely stall the game.